app/queries/servers/preference.ts

```diff
@@ -46,3 +47,21 @@ export const queryThemeForCurrentTeam = async (database: Database) => {
 
     return undefined;
 };
+
+export const deletePreferences = async (database: ServerDatabase, preferences: PreferenceType[]) => {
+    try {
+        const preparedModels: Model[] = [];
+        for await (const pref of preferences) {
+            const myPrefs = await queryPreferencesByCategoryAndName(database.database, pref.category, pref.name);
+            for (const p of myPrefs) {
+                preparedModels.push(p.prepareDestroyPermanently());
+            }
+        }
+        if (preparedModels.length) {
+            await database.operator.batchRecords(preparedModels);
+        }
+    } catch (error) {
+        // Record not found, do nothing
```

Comment:
Perhaps have this function return a boolean?

Reply:
How about returning the following?;;

Reply:
Just a boolean is fine, false in the catch block, true otherwise